src/App/Pages/Generator/GeneratorPage.xaml.cs

```diff
@@ -138,7 +143,14 @@ private async void Close_Clicked(object sender, EventArgs e)
         {
             if (DoOnce())
             {
-                await Navigation.PopModalAsync();
+                if (_fromExtension)
+                {
+                    _deviceActionService.ClosePopUpPage();
+                }
+                else
+                {
+                    await Navigation.PopModalAsync();
+                }
             }
```

Comment:
Move the whole Close to a command in the ViewModel and bind to it. Thus move the logic to the ViewModel as well and remove the initialization of the `_deviceActionService` from here.